src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs

```diff
@@ -142,12 +142,16 @@ public async Task ContainerIndicesAreUpdated()
 			Assert.AreEqual(container0, containerItem);
 		}
 
-
 #if HAS_UNO
 		[TestMethod]
 		[RunsOnUIThread]
+		[Ignore("Disabled because of the invalid assumption.")]
 		public async Task ContainerIndicesAreUpdated_OnRemoveAndAdd()
 		{
+			// This test used to pass. Because the IndexForItemContainer was never changed during the removal and re-insertion.
+			// In #10281, LVB has been updated to make sure the LVItem(container) is immediately clean up (which sets its IndexForItemContainer to -1) on removal.
+			// It revealed that the index is not updated for insertion. In fact, that and the MaterializedContainers are only updated on measure pass.
```

Comment:
This would mean that it could break drag and drop? What do you think @dr1rrb? 

Related: https://github.com/unoplatform/uno/pull/8406

Reply:
Visual items should not be removed while drag and dropping, so it should not break the D&D ... but we should validated. We do have UI tests, but some has been disabled as they was flaky (such kind of tests are difficult to stabilize with Xamarin UI tests)

BTW if the test was not testing the right thing (does it pass on WinUI ???), I would just update it instead of disabling it. I would assume "Add" and "removed" indices of other items should still be updated.

Reply:
This is a test that limited to uno. MaterializedContainers & IndexForItemContainerProperty are `internal` from uno.
I can try to poke around winui with reflection to see if we got anything similar.

Reply:
You can get index of an item / container by:
1. Get the `ItemsControl` for a random item using https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.itemscontrol.getitemsowner?view=winrt-22621 
2. The get the index of that item using https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.itemscontrol.indexfromcontainer?view=winrt-22621

Not sure it's enough to run the test on Windows, but the idea is that even if the test was enforcing some invalid cases, it seems really legit that we validate that adding/removing items in the list will update the index of all items (including the added / removed one). So if we are able to validate those cases instead of just disabling the whole test, it would be great.

Reply:
We dont have anything comparable on windows for MaterializedContainers. On windows, if we inspect the visual tree during removal and insertion of item, we will see that the LVItem (container) is removed immediately from the tree, and re-insertion does nothing until remeasured.
The test has been updated to reflect the current behavior of uno.